gateway/crates/federated-server/src/server/gateway.rs

```diff
@@ -161,11 +170,29 @@ pub(super) async fn generate(
                 tls,
             };
 
-            RedisRateLimiter::runtime(global_config, rate_limiting_configs)
+            match config_path {
+                Some(path) if config_hot_reload => {
+                    hot_reload::ConfigWatcher::new(path, tx).watch()?;
```

Comment:
:zap: **Suggestion** 
Consider handling the case where `hot_reload::ConfigWatcher::new(path, tx).watch()?` fails, to prevent the application from crashing.